src/internal/serviceaccount/process.go

```diff
@@ -149,8 +149,12 @@ func ensureServiceAccountApiKeyStep(step *StepContext) error {
 			return err
 		}
 
-		if step.HasClusterApiKey(clusterAccess) {
-			return nil
+		hasKey, err := step.HasClusterApiKey(clusterAccess)
+		if err != nil {
+			return err
+		}
+		if hasKey {
```

Comment:
This empty if-statement seem like an oversight?
What should it contain?

Reply:
It is supposed to return, so that we dont try to create a new key. I have fixed it